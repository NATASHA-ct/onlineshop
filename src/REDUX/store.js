/* eslint-disable */
import { configureStore } from "@reduxjs/toolkit";
import productReducer from "./products/productsSlice";
import productDetailsReducer from "./products/productsdetailsSlice";

const store = configureStore({
  reducer: {
    products: productReducer,
    productDetail: productDetailsReducer,
  },
});

export default store;
