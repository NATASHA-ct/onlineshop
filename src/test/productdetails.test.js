import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../REDUX/store';
import ProductDetails from '../COMPONENTS/PRODUCTSDETAILS/productdetails';

describe('to test the ProductDetails components', () => {
  test('matches the ProductDetails DOM snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <ProductDetails />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
