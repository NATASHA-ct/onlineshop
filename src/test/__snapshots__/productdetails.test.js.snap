// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to test the ProductDetails components matches the ProductDetails DOM snapshot 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  >
    <button
      className="goback-btn"
    >
       Click here to go back
    </button>
  </a>
  <div
    className="product-container"
  >
    <img
      className="prod-img"
    />
    <h1
      className="product-title"
    />
    <div>
      <p
        className="product-price"
      >
        $ 
      </p>
    </div>
    <p
      className="product-descrip"
    />
  </div>
</div>
`;

exports[`to test the Productcard components matches the Productcard DOM snapshot 1`] = `
<a
  href="/product/undefined"
  onClick={[Function]}
>
  <div
    className="card"
  >
    <div
      className="card-container"
    >
      <img
        className="card-img"
        height="200px"
        width="200px"
      />
      <div
        className="card-details"
      >
        <h1
          className="card-title"
        />
        <p
          className="card-price"
        >
          $ 
        </p>
      </div>
    </div>
  </div>
</a>
`;
